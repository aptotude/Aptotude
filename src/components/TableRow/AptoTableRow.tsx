import React from 'react';
import { StandardTypes } from '../../utils/standardTypes';
import classNames from 'classnames';
import './aptoTableRow.scss';

const COMPONENT_PREFIX = 'AptoTableRow';

export interface AptoTableRowDisplayProps extends StandardTypes {
  highlight?: 'danger' | 'primary' | 'warning' | 'success' | undefined;
}

export class AptoTableRow extends React.Component<
  AptoTableRowDisplayProps
> {
  public render() {
    const { highlight, children, ...rest } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      highlight && `${COMPONENT_PREFIX}--${highlight}`
    );

    return (
      <tr className={classes} {...rest}>
        {children}
      </tr>
    );
  }
}
