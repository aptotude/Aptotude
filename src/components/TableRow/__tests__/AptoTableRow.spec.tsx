import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoTableRow } from '../AptoTableRow';

afterEach(() => {
  cleanup();
});

describe('AptoTableRow', () => {
  it('Renders Component', () => {
    const { container } = render(
      <table>
        <tbody>
          <AptoTableRow>
            <td>hi</td>
          </AptoTableRow>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders extra class names', () => {
    const { container } = render(
      <table>
        <tbody>
          <AptoTableRow className="foo">
            <td>hi</td>
          </AptoTableRow>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('highlight', () => {
    it('shows danger', () => {
      const { container } = render(
        <table>
          <tbody>
            <AptoTableRow highlight="danger">
              <td>hi</td>
            </AptoTableRow>
          </tbody>
        </table>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('shows primary', () => {
      const { container } = render(
        <table>
          <tbody>
            <AptoTableRow highlight="primary">
              <td>hi</td>
            </AptoTableRow>
          </tbody>
        </table>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('shows warning', () => {
      const { container } = render(
        <table>
          <tbody>
            <AptoTableRow highlight="warning">
              <td>hi</td>
            </AptoTableRow>
          </tbody>
        </table>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    it('shows success', () => {
      const { container } = render(
        <table>
          <tbody>
            <AptoTableRow highlight="success">
              <td>hi</td>
            </AptoTableRow>
          </tbody>
        </table>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
