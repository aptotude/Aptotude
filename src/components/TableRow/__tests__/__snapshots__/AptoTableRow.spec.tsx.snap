// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoTableRow Renders Component 1`] = `
<table>
  <tbody>
    <tr
      class="AptoTableRow"
    >
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableRow Renders extra class names 1`] = `
<table>
  <tbody>
    <tr
      class="foo"
    >
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableRow highlight shows danger 1`] = `
<table>
  <tbody>
    <tr
      class="AptoTableRow AptoTableRow--danger"
    >
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableRow highlight shows primary 1`] = `
<table>
  <tbody>
    <tr
      class="AptoTableRow AptoTableRow--primary"
    >
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableRow highlight shows success 1`] = `
<table>
  <tbody>
    <tr
      class="AptoTableRow AptoTableRow--success"
    >
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`AptoTableRow highlight shows warning 1`] = `
<table>
  <tbody>
    <tr
      class="AptoTableRow AptoTableRow--warning"
    >
      <td>
        hi
      </td>
    </tr>
  </tbody>
</table>
`;
